otherlibs/stdune/src/compact_position.ml

```diff
@@ -0,0 +1,134 @@
+module Position = struct
+  (* We encode the position in three 21-bit chunks: [cnum][lnum][bol] *)
+  type t = int
```

Comment:
That isn't going to work 32-bit platforms. Maybe that's not a big deal because Dune was already broken on them?